### PR TITLE
Adds saveNewFederation tests

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds saveNewFederation tests

### CODE REVIEW

**Review by wilmerrootstock — APPROVED** [review]
Well Done!

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -314,94 +314,32 @@ void getPendingFederation_nullPendingFederationInStorage_returnsNull() {
...
-            ConsensusRule.RSKIP201
-        ).forBlock(0);
-        testSaveNewFederation(
-            STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION,
-            createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)
-        );
-    }
+        ActivationConfig.ForBlock activationsPostPapyrus = ActivationConfigsForTest.papyrus200().forBlock(0L);
+        ActivationConfig.ForBlock activationsPostIris = ActivationConfigsForTest.iris300().forBlock(0L);
+        ActivationConfig.ForBlock activationsPostHop = ActivationConfigsForTest.hop400().forBlock(0L);
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:321]:
```suggestion
        ActivationConfig.ForBlock papyrusActivations = ActivationConfigsForTest.papyrus200().forBlock(0L);
        ActivationConfig.ForBlock irisActivations = ActivationConfigsForTest.iris300().forBlock(0L);
        ActivationConfig.ForBlock hopActivations = ActivationConfigsForTest.hop400().forBlock(0L);
```

**Review by nathanieliov — APPROVED** [review]
Just a suggestion

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -311,94 +311,26 @@ void getPendingFederation_nullPendingFederationInStorage_returnsNull() {
...
-    void saveNewFederation_after_RSKIP123_should_save_standard_multisig_fed_format() {
-        activations = ActivationConfigsForTest.only(ConsensusRule.RSKIP123).forBlock(0);
-        testSaveNewFederation(
-            STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION,
-            createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION)
-        );
-    }
+        Federation standardFederation = createFederation(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION);
+        Federation nonStandardFederation = createFederation(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION);
+        Federation ps2hErpFederation = createFederation(P2SH_ERP_FEDERATION_FORMAT_VERSION);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Let's use the different Federation builders we already have instead

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**